bboxes labelling/additional_channels_extraction.py

```diff
@@ -0,0 +1,313 @@
+    -------
+    gray_frame : np.array
+        grayscaled input frame
+    blurred_frame : np.array
+        Gaussian-blurred grayscaled input frame
+    """
+
+    # convert the frame to grayscale frame
+    gray_frame = cv2.cvtColor(frame, cv2.COLOR_BGR2GRAY)
+    # apply Gaussian blurring
+    blurred_frame = cv2.GaussianBlur(gray_frame, kernel_size, sigmax)
+
+    return gray_frame, blurred_frame
+
+
+def compute_mean_and_max_abs_blurred_frame(cap, kernel_size, sigmax):
+    """Compute the mean blurred frame and the maximum absolute-value
+    blurred frame for a video capture cap
+
+    Parameters
+    ----------
+    cap : _type_
+        OpenCV VideoCapture object
+    kernel_size : list
+        kernel size for GaussianBlur
+    sigmax : int
+        Standard deviation in the X direction of the Gaussian kernel
+
+    Returns
+    -------
+    mean_blurred_frame : np.array
+        mean of all blurred frames in the video
+    max_abs_blurred_frame : np.array
+        pixelwise max absolute value across all blurred frames in the video
+    """
+    frame_counter = 0
+
+    # get image size
+    width = cap.get(cv2.CAP_PROP_FRAME_WIDTH)
+    height = cap.get(cv2.CAP_PROP_FRAME_HEIGHT)
+
+    # initialise array for mean blurred frame
+    mean_blurred_frame = np.zeros((int(height), int(width)))
+
+    # initialise array for max_abs_blurred_frame
+    max_abs_blurred_frame = np.zeros((int(height), int(width)))
+
+    while True:
+        ret, frame = cap.read()
+        if not ret:
+            # Break the loop if no more frames to read
+            print(f"Cannot read the frame{frame_counter}. Exiting...")
+            break
+
+        # apply transformations to the frame
+        _, blurred_frame = apply_grayscale_and_blur(frame, kernel_size, sigmax)
+
+        # accumulate blurred frames
+        mean_blurred_frame += blurred_frame
+
+        # accumulate max absolute values
+        max_abs_blurred_frame = np.maximum(max_abs_blurred_frame, abs(blurred_frame))
+
+        # update frame counter
+        frame_counter += 1
+
+    # compute the mean
+    mean_blurred_frame = mean_blurred_frame / frame_counter
+
+    return mean_blurred_frame, max_abs_blurred_frame
+
+
+def compute_background_subtracted_frame(
+    blurred_frame, mean_blurred_frame, max_abs_blurred_frame
+):
+    """Compute the background subtracted frame for the
+    input blurred frame, given the mean and max absolute frames of
+    its corresponding video
+
+    Parameters
+    ----------
+    blurred_frame : np.array
+        Gaussian-blurred grayscaled input frame
+    mean_blurred_frame : np.array
+        mean of all blurred frames in the video
+    max_abs_blurred_frame : np.array
+        pixelwise max absolute value across all blurred frames in the video
+
+    Returns
+    -------
+    background_subtracted_frame : np.array
+        normalised difference between the blurred frame f and
+        the mean blurred frame
+    """
+    background_subtracted_frame = (
```

Comment:
do you know why do we need to do +1 and then divide by 2? 🤔 I know they do it in the reference also but not sure why.... Is the +1 to avoid zeros? (why?)